synthesizer/src/snark/batch/mod.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use anyhow::anyhow;
+
+use super::*;
+
+mod bytes;
+mod parse;
+mod serialize;
+
+use std::collections::BTreeMap;
+
+#[derive(Clone, PartialEq, Eq)]
+pub enum Key<N: Network> {
+    ProvingKey(ProvingKey<N>),
+    VerifyingKey(VerifyingKey<N>),
+}
+
+#[derive(Clone)]
```

Comment:
```suggestion
#[derive(Clone, Copy, PartialEq, Eq)]
```
seems reasonable to extend these to such a simple enum

Reply:
Done